### PR TITLE
fix(zookeeper): validate clustering inputs

### PR DESCRIPTION
**Pull request by ReubenBond**
## Problem
The ZooKeeper clustering provider had ten CA1062 findings across its membership and gateway option boundaries. Invalid inputs could fail after path composition or serialization instead of at the public boundary.

## Solution
- Validate the ten analyzer-reported inputs with exact `ArgumentNullException.ParamName` values.
- Enforce CA1062 for the complete `Orleans.Clustering.ZooKeeper` project.
- Add service-independent tests covering synchronous null contracts, literal ZooKeeper path composition, gateway settings, and the persisted membership JSON schema and legacy `InstanceName` compatibility.

The ZooKeeper transaction ordering, version checks, gateway filtering, connection behavior, and nullable hosting configuration delegate behavior remain unchanged.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/orleans/pull/11138)

### CODE REVIEW

**Comment by ReubenBond** [comment]
The Azure Storage provider tests (net8.0) failure is the known unrelated SMS_StreamRel_AllSilosRestart_PubSubCounts duplicate-publisher flake tracked in #11033. The ZooKeeper provider jobs passed on both net8.0 and net10.0, and this PR does not change streaming or Azure Storage code.

File: src/Orleans.Clustering.ZooKeeper/ZooKeeperBasedMembershipTable.cs [hunk 1]
```diff
@@ -63,11 +63,17 @@ public partial class ZooKeeperBasedMembershipTable : IMembershipTable
...
+        /// <paramref name="membershipTableOptions"/> or <paramref name="clusterOptions"/> is <see langword="null"/>.
+        /// </exception>
         public ZooKeeperBasedMembershipTable(
             ILogger<ZooKeeperBasedMembershipTable> logger,
             IOptions<ZooKeeperClusteringSiloOptions> membershipTableOptions,
             IOptions<ClusterOptions> clusterOptions)
         {
+            ArgumentNullException.ThrowIfNull(membershipTableOptions);
+            ArgumentNullException.ThrowIfNull(clusterOptions);
+
```
[the remark this answers is not in the document]
Comment (ReubenBond) [reply src/Orleans.Clustering.ZooKeeper/ZooKeeperBasedMembershipTable.cs:78]: Fixed in `f727a34926`. The membership-table constructor now validates `logger` before using it, its XML exception contract includes the logger parameter, and focused coverage verifies `ParamName == "logger"`.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: src/Orleans.Clustering.ZooKeeper/ZooKeeperGatewayListProvider.cs [hunk 2]
```diff
@@ -34,12 +34,19 @@ public class ZooKeeperGatewayListProvider : IGatewayListProvider
...
+        /// </exception>
         public ZooKeeperGatewayListProvider(
             ILogger<ZooKeeperGatewayListProvider> logger,
             IOptions<ZooKeeperGatewayListProviderOptions> options,
             IOptions<GatewayOptions> gatewayOptions,
             IOptions<ClusterOptions> clusterOptions)
         {
+            ArgumentNullException.ThrowIfNull(options);
+            ArgumentNullException.ThrowIfNull(gatewayOptions);
+            ArgumentNullException.ThrowIfNull(clusterOptions);
```
[the remark this answers is not in the document]
Comment (ReubenBond) [reply src/Orleans.Clustering.ZooKeeper/ZooKeeperGatewayListProvider.cs:50]: Fixed in `f727a34926`. The gateway-list constructor now validates `logger`, documents the boundary contract, and has focused coverage for the exact parameter name.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: src/Orleans.Clustering.ZooKeeper/ZooKeeperBasedMembershipTable.cs [hunk 3]
```diff
@@ -210,8 +222,14 @@ public Task<bool> InsertRow(MembershipEntry entry, TableVersion tableVersion)
...
         /// <param name="tableVersion">The new TableVersion for this table, along with its etag.</param>
         /// <returns>True if the update operation succeeded and false otherwise.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="entry"/> or <paramref name="tableVersion"/> is <see langword="null"/>.
+        /// </exception>
         public Task<bool> UpdateRow(MembershipEntry entry, string etag, TableVersion tableVersion)
         {
+            ArgumentNullException.ThrowIfNull(entry);
+            ArgumentNullException.ThrowIfNull(tableVersion);
+
```
[the remark this answers is not in the document]
Comment (ReubenBond) [reply src/Orleans.Clustering.ZooKeeper/ZooKeeperBasedMembershipTable.cs:236]: Fixed in `f727a34926`. `UpdateRow` now validates `etag` explicitly before parsing, documents the null contract, and has focused coverage verifying `ParamName == "etag"`.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).
